dotCMS/src/main/java/com/dotcms/ema/EMAWebInterceptor.java

```diff
@@ -0,0 +1,173 @@
+ * to render the EDIT_MODE request in context.
+ *
+ * More info on how EMA works https://github.com/dotcms-plugins/com.dotcms.ema#dotcms-edit-mode-anywhere---ema
+ */
+public class EMAWebInterceptor  implements WebInterceptor{
+
+    public  static final String      PROXY_EDIT_MODE_URL_VAR = "proxyEditModeURL";
+    private static final String      API_CALL                = "/api/v1/page/render";
+    public static final String EMA_APP_CONFIG_KEY = "dotema-config";
+    private static final ProxyTool   proxy                   = new ProxyTool();
+
+
+    @Override
+    public String[] getFilters() {
+        return new String[] {
+                API_CALL + "*"
+        };
+    }
+
+    @Override
+    public Result intercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        final Host currentHost = WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request);
+
+        if (!this.existsConfiguration(currentHost.getIdentifier())) {
+            return Result.NEXT;
+        }
+
+        final Optional<String> proxyUrl = proxyUrl(currentHost);
+        final PageMode mode             = PageMode.get(request);
+
+        if (!proxyUrl.isPresent() || mode == PageMode.LIVE) {
+            return Result.NEXT;
+        }
+
+        Logger.info(this.getClass(), "GOT AN EMA Call --> " + request.getRequestURI());
+
+        return new Result.Builder().wrap(new MockHttpCaptureResponse(response)).next().build();
+    }
+
+
+    @Override
+    public boolean afterIntercept(final HttpServletRequest request, final HttpServletResponse response) {
+
+        try {
+
+            if (response instanceof MockHttpCaptureResponse) {
+
+                final Host currentHost = WebAPILocator.getHostWebAPI().getCurrentHostNoThrow(request);
+                final Optional<String> proxyUrl            = proxyUrl(currentHost);
+                final MockHttpCaptureResponse mockResponse = (MockHttpCaptureResponse)response;
+                final String postJson                      = new String(mockResponse.getBytes());
+                final JSONObject json                      = new JSONObject(postJson);
+                final Map<String, String> params           = ImmutableMap.of("dotPageData", postJson);
+                
+                Logger.info(this.getClass(), "Proxying Request --> " + proxyUrl.get());
+
+                final StringBuilder responseStringBuilder = new StringBuilder();
+                final ProxyResponse pResponse = proxy.sendPost(proxyUrl.get(), params);
+
+                if (pResponse.getResponseCode() == 200) {
+                    responseStringBuilder.append(pResponse.getResponse());
+                }else {
+                    responseStringBuilder.append("<html><body>")
+                    .append("<h3>Unable to connect with the rendering engine</h3>")
+                    .append("<br><div style='display:inline-block;width:80px'>Trying: </div><b>" + proxyUrl.get()  + "</b>")
```

Comment:
Codacy found an issue: [StringBuffer (or StringBuilder).append is called 2 consecutive times with literals. Use a single append with a single combined String.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5922984)